lib/active_merchant/versionable.rb

```diff
@@ -0,0 +1,29 @@
+module ActiveMerchant
+  module Versionable
+    def self.included(base)
+      if base.respond_to?(:class_attribute)
+        base.class_attribute :versions, default: {}
+        base.extend(ClassMethods)
+      end
+    end
+
+    module ClassMethods
+      def inherited(subclass)
+        super
+        subclass.versions = {}
+      end
+
+      def version(version, feature = :default_api)
+        versions[feature] = version
+      end
+
+      def fetch_version(feature = :default_api)
```

Comment:
Also this one, how is it used? What is the difference between this one and the one in line 25?

Reply:
The `fetch_version` method inside ClassMethods is intended to be used outside of instance methods.

For example, some gateways create `live_url` and `test_url` at the top of the file without creating an instance, so you can do something like this:

```ruby
module ActiveMerchant # :nodoc:
  module Billing # :nodoc:
    class ForteGateway < Gateway
      version 'v2'

      self.test_url = "https://sandbox.forte.net/api/#{fetch_version}"
      self.live_url = "https://api.forte.net/#{fetch_version}"

      # other methods ...
    end
  end
end
```

The `fetch_version` method outside ClassMethods is intended to be used inside of instance methods. There are a couple of examples of this in this PR, you can see it being used for the Adyen gateway and GlobalCollect gateway.